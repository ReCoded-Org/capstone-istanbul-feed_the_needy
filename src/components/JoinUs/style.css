.joinUsSection {
  display: flex;
  background-color: #48555e;
}

.joinUsSectionImg {
  max-width: 21em;
}

@media only screen and (max-width: 943px) {
  .joinUsSectionImg {
    display: none;
  }
}

.joinUsSectionBtn {
  margin-top: 1em;
  width: 12em;
  height: 4em;
  color: #48555e;
  background-color: #f4f6f6;
  border-width: 0em;
  border-radius: 0em;
}
