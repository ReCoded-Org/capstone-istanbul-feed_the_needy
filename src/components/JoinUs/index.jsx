import React from "react";
import { Typography, Image, Button } from "antd";
import { useTranslation } from "react-i18next";
import JoinHands from "./images/joinHands.png";
import "./style.css";

const JoinUs = ({ isTesting }) => {
  const { t } = useTranslation();
  const { Title, Paragraph } = Typography;

  return (
    <div className="joinUsSection">
      <Image className="joinUsSectionImg" src={JoinHands} alt="JoinHands" />
      <div className="joinUsSectionDescription">
        <div style={{ margin: "1em" }}>
          <Title style={{ color: "#f4f6f6" }} level={2}>
            {t("joinUs.title")}
          </Title>
          <Paragraph style={{ color: "#f4f6f6" }}>
            {t("joinUs.description")}
          </Paragraph>
          {isTesting ? (
            <button type="button">Create account</button>
          ) : (
            <Button className="joinUsSectionBtn" type="primary">
              {t("joinUs.button")}
            </Button>
          )}
        </div>
      </div>
    </div>
  );
};

export default JoinUs;
