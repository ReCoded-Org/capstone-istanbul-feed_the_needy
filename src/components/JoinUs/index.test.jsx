import React from "react";
import renderer from "react-test-renderer";
import JoinUs from "./index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const isTesting = true;

it("JoinUs renders correctly", () => {
  const tree = renderer.create(<JoinUs isTesting={isTesting} />).toJSON();
  expect(tree).toMatchSnapshot();
});
