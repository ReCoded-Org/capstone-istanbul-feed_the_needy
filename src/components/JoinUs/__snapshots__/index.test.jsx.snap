// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinUs renders correctly 1`] = `
<div
  className="joinUsSection"
>
  <div
    className="ant-image joinUsSectionImg"
    onClick={[Function]}
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="JoinHands"
      className="ant-image-img"
      onError={[Function]}
      onLoad={[Function]}
      src="joinHands.png"
    />
  </div>
  <div
    className="joinUsSectionDescription"
  >
    <div
      style={
        Object {
          "margin": "1em",
        }
      }
    >
      <h2
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": null,
            "color": "#f4f6f6",
          }
        }
      >
        joinUs.title
      </h2>
      <div
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": null,
            "color": "#f4f6f6",
          }
        }
      >
        joinUs.description
      </div>
      <button
        type="button"
      >
        Create account
      </button>
    </div>
  </div>
</div>
`;
